src/main/java/frc/robot/commands/SwerveJoystickCommand.java

```diff
@@ -0,0 +1,123 @@
+
+    this.xSpdFunction = frontBackFunction;
+    this.ySpdFunction = leftRightFunction;
+    this.turningSpdFunction = turningSpdFunction;
+    this.speedIncreaseControlFunction = speedIncreaseControlFunction;
+    this.speedDecreaseControlFunction = speedDecreaseControlFunction;
+    this.xLimiter = new SlewRateLimiter(Constants.Swerve.kTeleDriveMaxAccelerationUnitsPerSecond);
+    this.yLimiter = new SlewRateLimiter(Constants.Swerve.kTeleDriveMaxAccelerationUnitsPerSecond);
+    this.turningLimiter =
+        new SlewRateLimiter(Constants.Swerve.kTeleDriveMaxAngularAccelerationUnitsPerSecond);
+    this.swerveDrivetrain = csd;
+
+    // Adds the subsystem as a requirement (prevents two commands from acting on subsystem at once)
+    addRequirements(swerveDrivetrain);
+  }
+
+  @Override
+  public void initialize() {}
+
+  @Override
+  public void execute() {
+    // 1. Get real-time joystick inputs
+    double xSpeed = xSpdFunction.get(); // xSpeed is actually front back (front +, back -)
+    double ySpeed = ySpdFunction.get(); // ySpeed is actually left right (left +, right -)
+    double turningSpeed =
+        turningSpdFunction.get(); // turning speed is (anti-clockwise +, clockwise -)
+
+    // 2. Normalize inputs
+    double length = xSpeed * xSpeed + ySpeed * ySpeed; // acutally length squared
+    if (length > 1d) {
+      length = Math.sqrt(length);
+      xSpeed /= length;
+      ySpeed /= length;
+    }
+
+    // 3. Apply deadband
+    xSpeed = Math.abs(xSpeed) > Constants.OI.kLeftJoystickDeadband ? xSpeed : 0.0;
+    ySpeed = Math.abs(ySpeed) > Constants.OI.kLeftJoystickDeadband ? ySpeed : 0.0;
+    turningSpeed =
+        Math.abs(turningSpeed) > Constants.OI.kRightJoystickDeadband ? turningSpeed : 0.0;
+
+    // 4. Make the driving smoother
+    // Speed control
+    double driveSpeed =
+        (Constants.Swerve.kTeleDriveMaxPercentSpeed - Constants.Swerve.kTeleDriveMinPercentSpeed)
+                * (speedIncreaseControlFunction.get() - speedDecreaseControlFunction.get())
+            + Constants.Swerve.kTeleDriveMinPercentSpeed;
+
+    // Applies slew rate limieter
+    xSpeed =
+        xLimiter.calculate(xSpeed) * driveSpeed * Constants.Swerve.kPhysicalMaxSpeedMetersPerSecond;
+    ySpeed =
+        yLimiter.calculate(ySpeed) * driveSpeed * Constants.Swerve.kPhysicalMaxSpeedMetersPerSecond;
+    turningSpeed =
+        turningLimiter.calculate(turningSpeed)
+            * driveSpeed
+            * Constants.Swerve.kPhysicalMaxAngularSpeedRadiansPerSecond;
+
+    // Final values to apply to drivetrain
+    final double x = xSpeed;
+    final double y = ySpeed;
+    final double turn = turningSpeed;
+
+    // 5. Applying the drive request on the swerve drivetrain
+    // Uses SwerveRequestFieldCentric (from java.frc.robot.util to apply module optimization)
+    final SwerveRequest.FieldCentric drive =
+        new SwerveRequestFieldCentric()
+            // .withDeadband(Constants.kPhysicalMaxSpeedMetersPerSecond *
```

Comment:
remove these comments, applying deadband through controller already (a controller deadband seems nicer conceptually to me than a speed deadband, so I think we can safely remove these comments)

Reply:
Agreed